baremetal/baremetalmachine_manager.go

```diff
@@ -837,3 +849,53 @@ func (m *MachineManager) SetProviderID(providerID string) {
 	m.BareMetalMachine.Spec.ProviderID = &providerID
 	m.BareMetalMachine.Status.Ready = true
 }
+
+// SetOwnerRef adds an ownerreference to this baremetal machine
+func (m *MachineManager) SetOwnerRef(refList []metav1.OwnerReference, controller bool) []metav1.OwnerReference {
+	index, err := m.FindOwnerRef(refList)
+	if err != nil {
+		refList = append(refList, metav1.OwnerReference{
+			APIVersion: m.BareMetalMachine.APIVersion,
+			Kind:       m.BareMetalMachine.Kind,
+			Name:       m.BareMetalMachine.Name,
+			UID:        m.BareMetalMachine.UID,
+			Controller: pointer.BoolPtr(controller),
+		})
+	} else {
+		refList[index].UID = m.BareMetalMachine.UID
+		refList[index].Controller = pointer.BoolPtr(controller)
+	}
+	return refList
+}
+
+// DeleteOwnerRef removes the ownerreference to this baremetal machine
+func (m *MachineManager) DeleteOwnerRef(refList []metav1.OwnerReference) []metav1.OwnerReference {
+	if len(refList) == 0 {
+		return refList
+	}
+	index, err := m.FindOwnerRef(refList)
+	if err == nil {
+		if len(refList) == 1 {
+			return []metav1.OwnerReference{}
+		}
+		refListLen := len(refList) - 1
+		refList[index] = refList[refListLen]
+		return (m.DeleteOwnerRef(refList[:refListLen-1]))
+	}
+	return refList
+}
+
+// FindOwnerRef checks if an ownerreference to this baremetal machine exists
+// and returns the index
+func (m *MachineManager) FindOwnerRef(refList []metav1.OwnerReference) (int, error) {
+	for i := 0; i < len(refList); i++ {
```

Comment:
Is there a reason not to use range() here?

```suggestion
	for i, curOwnerRef := range(refList) {
```

Reply:
no, indeed. I will fix it